ParkingApplication.api/Repositories/AccountRepository.cs

```diff
@@ -1,45 +1,40 @@
 using System.Data.SqlClient;
 using ParkingApplication.Api.Interfaces;
 using ParkingApplication.Api.Models;
+using System.Collections.Generic;
 
-namespace Repositories
-{
+namespace Repositories;
     public class AccountRepository : IQueryRepository<AccountModel>
 {
-private readonly SqlConnection _connection;
-
-public AccountRepository(SqlConnection connection)
-{
-    _connection = connection;
-}
+    private readonly IDbContext _dbContext;
 
-public List<AccountModel> GetAll()
-{
-    var accounts = new List<AccountModel>();
+    public AccountRepository(IDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
 
-    try
+    public List<AccountModel> GetAll()
     {
-        _connection.Open();
-        var command = new SqlCommand("SELECT * FROM Account", _connection);
+        var accounts = new List<AccountModel>();
 
-        using (var reader = command.ExecuteReader())
+        using (var connection = _dbContext.GetConnection())
```

Comment:
still creating new connection for each repo call. We need the connection to be already created where we register the context as scoped and then check in the repo if we need to close it or not (this is business logic so the service knows if it needs to be closed or not and it will send a parameter to the repo - by default repo should have that flag enabled to close the connection because usually from the service you have only one call to the repo, but sometimes you call the repo twice and you want to reuse the same connection, so the service will decide the flag to be false and the repo will not close the connection yet)